### PR TITLE
WIP: Support querying & converting 1 row of data.

### PR DESCRIPTION
**Pull request by jtv**
This makes it convenient to query either exactly 1 row of data (using
`query1()`), or at most 1 row of data (using `query01()`), and convert
the results straight to the desired client-side types.

The version that allows an empty result returns a `std::optional`, which
obviously is empty if the query produced no data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,10 +113,45 @@ Here is a simple example program to get you going, with full error handling:
...
             W.exec0("UPDATE employee SET salary = salary*2");
 
+            // Easy way to query a value from the database.
+            int my_salary = W.query_value<int>(
+                "SELECT salary FROM employee WHERE name = 'Me'");
+            std::cout << "I now earn " << my_salary << ".\n";
+
+            // Or, query one whole row.  This will throw an exception unless
+            // the result contains exactly 1 row.
+            auto top_name, top_salary = W.query1<std::string, int>(
```
Comment (KayEss) [inline README.md:145]: I thought this syntax had to be `auto [top_name, top_salary] = ...`

File: include/pqxx/transaction_base.hxx [hunk 2]
```diff
@@ -389,6 +400,42 @@ public:
...
 
+  /// Perform query returning exactly one row, and convert its fields.
+  /** This is a convenient way of querying one row's worth of data, and
+   * converting its fields to a tuple of the C++-side types you specify.
+   *
+   * @throw unexpected_rows If the query did not return exactly 1 row.
+   * @throw usage_error If the number of columns in the result does not match
+   * the number of fields in the tuple.
+   */
+  template<typename... TYPE> std::tuple<TYPE...> query1(zview query)
```
Comment (KayEss) [inline include/pqxx/transaction_base.hxx:411]: I think you should have `inline` on these

Comment (KayEss) [reply include/pqxx/transaction_base.hxx:411]: And presumably `[[nodiscard]]`

File: README.md (hunk 1 above)
Comment (jtv) [reply README.md:145]: Oops yes, you're right!  No idea how that happened, but it's probably got something to do with me typing it wrong.  Thanks.

File: include/pqxx/transaction_base.hxx (hunk 2 above)
Comment (jtv) [reply include/pqxx/transaction_base.hxx:411]: They're defined inline in the class, so the `inline` is implicit.  But you're right, they ought to be `[[nodiscard]]`.

Comment (KayEss) [reply include/pqxx/transaction_base.hxx]: Ah, sorry. I missed that they were class members :-)

**Pull request merged**